docs: add reporter configuration ref and document HEARTBEAT_PERIOD behaviour (#314)
kubernetes-sigs/node-readiness-controller: docs/book/src/user-guide/concepts.md

Reviewer: Can you move default status below enforcement modes? As it has references to the latter?

Author: Updated

*   **Simplicity**: Good for simple "is this HTTP endpoint up?" checks without configuring external scripts.
*   **Direct Coupling**: Useful when you want the readiness reporting lifecycle of the component to strictly match the pod's lifecycle.

#### Optimizing node status writes

> On every check interval, the reporter compares the observed component health

Reviewer: ```suggestion
On every `CHECK_INTERVAL`, the reporter compares the health result it just
```

Author: Updated

*   **Simplicity**: Good for simple "is this HTTP endpoint up?" checks without configuring external scripts.
*   **Direct Coupling**: Useful when you want the readiness reporting lifecycle of the component to strictly match the pod's lifecycle.

> ### Reducing unnecessary node status writes

Reviewer: ```suggestion
#### Reducing unnecessary node status writes
```
I wonder if we can rephrase it in a way that doesn't read "unnecessary".

Author: Rephrased it to "Optimizing node status writes"

On every check interval, the reporter compares the health result it just

> observed against the condition already stored on the Node (`Status`,

Reviewer: ```suggestion
On every check interval, the reporter compares the observed component health against the condition already stored on the Node (`Status`,
```

Author: Updated

> `Reason`, and `Message`). If all three match what's already there, the

Reviewer: ```suggestion
`Reason`, and `Message`). If the condition matches these and differs only in `lastHeartbeatTime`, the
```
Note: confirm the field name!

Author: Confirmed. lastHeartbeatTime is the correct API/JSON field name (the Go struct field is LastHeartbeatTime)

> reporter skips the write entirely.

Reviewer: ```suggestion
reporter skips updating the readiness condition.
```

Author: Updated

This matters at scale: writing a Node's status forces the API server to
persist the entire Node object to `etcd`, even if only one condition field
would have changed. Skipping no-op writes avoids that cost across large

> fleets of nodes reporting on a short interval.

Reviewer: ```suggestion
This is preferred in large scale clusters. Skipping no-op writes helps scale by reducing the condition update load for the API server.
```

Author: Updated

To make sure a reporter with nothing new to add doesn't look stalled, it
still writes at least once every `HEARTBEAT_PERIOD` (default `5m`),
refreshing the condition's `lastHeartbeatTime` even when nothing changed.
If the health status *does* change, the reporter writes immediately,

> regardless of this timer.

Reviewer: ```suggestion
To ensure reliability, the reporter forces a refresh of the condition every `HEARTBEAT_PERIOD`, which is default to 5 minutes.
```

Author: Updated

> [!NOTE]
> This only reduces *writes*. The reporter still reads the Node object on
> every check interval to compare state, so this does not reduce the
> number of requests sent to the API server, only the number of updates

> > persisted to `etcd`.

Reviewer: ```suggestion
> number of read requests sent to the API server, only the number of updates
```

Author: Updated

See [Reporter Configuration](../reference/reporter-configuration.md) for the

> full list of environment variables, including `HEARTBEAT_PERIOD`.

Reviewer: ```suggestion
all supported configuration.
```

Author: Updated
